common/component/kafka/metadata.go

```diff
@@ -115,10 +115,10 @@ type KafkaMetadata struct {
 }
 
 // upgradeMetadata updates metadata properties based on deprecated usage.
-func (k *Kafka) upgradeMetadata(metadata map[string]string) (map[string]string, error) {
-	authTypeVal, authTypePres := metadata[authType]
-	authReqVal, authReqPres := metadata["authRequired"]
-	saslPassVal, saslPassPres := metadata["saslPassword"]
+func (k *Kafka) upgradeMetadata(meta map[string]string) (map[string]string, error) {
+	authTypeVal, authTypePres := meta[authType]
+	authReqVal, authReqPres := meta["authRequired"]
+	saslPassVal, saslPassPres := meta["saslPassword"]
```

Comment:
Please use `metadata.GetMetadataProperties` like I mentioned, it has to be case-insensitive.